pkg/apis/navigator/validation/cassandra.go

```diff
@@ -1,15 +1,59 @@
 package validation
 
 import (
+	"fmt"
 	"reflect"
+	"strings"
 
 	apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 
 	"github.com/jetstack/navigator/pkg/apis/navigator"
+	"github.com/jetstack/navigator/pkg/cassandra/version"
 )
 
+type versionSpecOp string
+
+const (
+	versionSpecGe versionSpecOp = ">="
+	versionSpecLt               = "<"
+)
+
+type versionSpec map[versionSpecOp]*version.Version
+
+var supportedCassandraVersions = versionSpec{
+	versionSpecGe: version.New("3.0.0"),
+	versionSpecLt: version.New("4.0.0"),
+}
+
+func (vs versionSpec) Check(v *version.Version) bool {
+	for op, opVersion := range vs {
+		switch op {
+		case versionSpecGe:
+			if v.LessThan(opVersion) {
+				return false
+			}
+		case versionSpecLt:
+			if opVersion.Equal(v) {
+				return false
+			}
+			if opVersion.LessThan(v) {
```

Comment:
We switch to testing `opVersion` instead of `v` here - can we be consistent for readability?

Reply:
I deleted this code and instead use github.com/hashicorp/go-version constraints.